### PR TITLE
Add convertor for Mem JSON export

### PR DESCRIPTION
**Pull request by maccman**

### CODE REVIEW

File: src/convertors/mem/mem-convertor.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      }
+
+      if (typeof note.created !== 'string') {
+        throw new MemConversionError('Note created must be a string')
+      }
+
+      if (typeof note.updated !== 'string') {
+        throw new MemConversionError('Note updated must be a string')
+      }
+    }
```
Comment (ocavue) [inline src/convertors/mem/mem-convertor.ts:104]: Why don't we use `zod` to do checking. We can apply more rules and less code (for example, the code here didn't check that every item in `note.tags` should be a string)

File: src/convertors/mem/mem-convertor.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    // Combine unique tags from markdown and mem
+    const tags = [...new Set([...markdownTags, ...memNote.tags])]
+
+    const note: ConvertedNote = {
+      id: `mem-${memNote.id}`,
+      html,
+      subject: memNote.title,
+      dailyAt: dailyDate?.getTime(),
+      backlinks,
```
[the remark this answers is not in the document]
Comment (maccman) [reply src/convertors/mem/mem-convertor.ts:64]: You are so good at reviews!

File: src/convertors/mem/mem-convertor.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      linkHost: this.linkHost,
+    })
+
+    const dailyDate = getDailyDate(memNote)
+
+    const updatedAt = parseTimestamp(memNote.updated)
+    const createdAt = parseTimestamp(memNote.created)
+
+    // Combine unique tags from markdown and mem
+    const tags = [...new Set([...markdownTags, ...memNote.tags])]
```
Comment (ocavue) [inline src/convertors/mem/mem-convertor.ts:57]: Let's slugify all note ids and tag, just in case that Mem's format doesn't fit our requirement

Comment (ocavue) [reply src/convertors/mem/mem-convertor.ts:57]: [slugify](https://github.com/sindresorhus/slugify)

Comment (maccman) [reply src/convertors/mem/mem-convertor.ts:59]: CleanShot 2023-05-08 at 10 13 03@2x

Not sure this is a great idea.

File: src/convertors/mem/mem-convertor.ts (hunk 1 above)
Comment (maccman) [reply src/convertors/mem/mem-convertor.ts]: Yeah, I'll switch it to Zod.

File: src/convertors/mem/mem-convertor.ts (hunk 3 above)
Comment (ocavue) [reply src/convertors/mem/mem-convertor.ts:59]: You can use the decamelize option in the lib

Comment (maccman) [reply src/convertors/mem/mem-convertor.ts:59]: The IDs look consistent to me. I think let's reserve Slugify for when we're using something like slugifying titles.

**Pull request merged**